### PR TITLE
Auto docs generation for gh-pages

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
The documentation is generated in the` gh-pages` branch in different folders depending on different events:

- push to the `main` branch ----> to the `dev` folder

- push to the `release*` branch ----> to the `next_release` folder

- create tags for a release ----> to the `tag_name` folder

### CODE REVIEW

File: scripts/install_dpctl.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+DPCTL_TARGET_VERSION=0.5.0rc2
+echo ++++++++++++++++++ Build DPCTL ${DPCTL_TARGET_VERSION} +++++++++++++++++++
+git clone --branch ${DPCTL_TARGET_VERSION} https://github.com/IntelPython/dpctl.git 
+
+cd dpctl
+
+git tag ${DPCTL_TARGET_VERSION}
+
+conda build conda-recipe/ --no-test -c ${ONEAPI_ROOT}/conda_channel
```
Comment (PokhodenkoSA) [inline scripts/install_dpctl.sh:15]: Here you build with packages from oneAPI. Why did not you use numba from this channel too?

**Review by PokhodenkoSA — CHANGES_REQUESTED** [review]

File: .github/workflows/build-sphinx.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
+name: Build Sphinx
+on:
+  push:
+    branches:
+      - main
+      - release*
+    tags:
+      - '[0-9].+**'
```
Comment (PokhodenkoSA) [inline .github/workflows/build-sphinx.yml:8]: Please find a better regexp to match tags.

File: .github/workflows/build-sphinx.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+jobs:
+  build-and-deploy:
+    name: Build and Deploy Docs
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.6.0
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply .github/workflows/build-sphinx.yml:16]: I used this workflow during testing. I will delete it, thank you .

File: .github/workflows/build-sphinx.yml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        with:
+          access_token: ${{ github.token }}
+      - name: Install Intel repository
+        run: |
+          wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
+          sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
+          rm GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
+          sudo add-apt-repository "deb https://apt.repos.intel.com/oneapi all main"
+          sudo apt-get update
+      - name: Update libstdc++-dev
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply .github/workflows/build-sphinx.yml:26]: Great question!
This is really not necessary.

File: .github/workflows/build-sphinx.yml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        run: |
+          sudo apt remove -y gcc-7 g++-7 gcc-8 g++-8 gcc-10 g++-10
+          sudo apt remove -y libstdc++-10-dev
+          sudo apt autoremove
+          sudo apt install --reinstall -y gcc-9 g++-9 libstdc++-9-dev
+      - name: Install Intel OneAPI
+        run: |
+          sudo apt-get install intel-oneapi-mkl                \
+                               intel-oneapi-mkl-devel          \
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply .github/workflows/build-sphinx.yml:35]: You are right. I will delete it.

File: .github/workflows/build-sphinx.yml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+                               intel-oneapi-dpcpp-cpp-compiler \
+                               intel-opencl-icd
+      # https://github.com/marketplace/actions/checkout
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply .github/workflows/build-sphinx.yml:38]: Ok. I will delete this line

File: .github/workflows/build-sphinx.yml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+      - name: Checkout repo
+        uses: actions/checkout@v2
+      # https://github.com/marketplace/actions/setup-miniconda
+      - name: Setup miniconda
+        uses: conda-incubator/setup-miniconda@v2
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply .github/workflows/build-sphinx.yml:43]: There are difficulties in using the default conda in github actions. It is better to install a сonda to escape all difficulties.

File: .github/workflows/build-sphinx.yml [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+          conda info
+          conda list
+      - name: Install numba-dppy dependencies
+        shell: bash -l {0}
+        run: |
+          conda install numpy cython cmake conda-build llvmdev llvmlite recommonmark sphinx sphinx_rtd_theme wheel
+          ./scripts/0.env.sh
+          git clone https://github.com/IntelPython/numba.git
+          cd numba
+          python setup.py develop
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply .github/workflows/build-sphinx.yml:60]: Intel channel has version 0.51.2 which is not compatible with numba-dppy.

File: .github/workflows/build-sphinx.yml [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+        id: capture_tag
+        run: |
+          echo "${GITHUB_REF#refs/heads/}"
+          echo "${GITHUB_REF#refs/tags/}"
+          echo ::set-output name=tag_number::${GITHUB_REF#refs/tags/}
+      - name: Build docs
+        shell: bash -l {0}
+        run: |
+          . /opt/intel/oneapi/setvars.sh
+          ./0.builddoc.sh
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply .github/workflows/build-sphinx.yml:72]: You are right. I will fix it.

File: scripts/install_dpctl.sh [hunk 10]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+THEDIR=$(dirname $(readlink -e ${BASH_SOURCE[0]}))
+
+. ${THEDIR}/0.env.sh
+
+DPCTL_TARGET_VERSION=0.5.0rc2
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply scripts/install_dpctl.sh:7]: I will remove this script. Dpctl will be installed from the intel channel with version 0.3.8

**Comment by vlad-perevezentsev** [comment]
Required changes:
1. Use cash to install OneApi
2. Correct use of conda.
3. Better pattern for tags.

**Pull request closed without merging**